lib/legacy/zstd_v06.c

```diff
@@ -3919,6 +3919,7 @@ ZBUFFv06_DCtx* ZBUFFv06_createDCtx(void)
     if (zbd==NULL) return NULL;
     memset(zbd, 0, sizeof(*zbd));
     zbd->zd = ZSTDv06_createDCtx();
+    if (zbd->zd==NULL) return NULL;
```

Comment:
Returning immediately here would result in a memory leak, because `zbd` has been allocated at that point.
Free `zbd`, using `ZBUFFv06_freeDCtx()`, before returning.

Reply:
Oh, indeed. Thanks a lot for the review, I will fix that.

Reply:
Done.